packages/web-api/src/WebClient.spec.ts

```diff
@@ -964,6 +971,76 @@ describe('WebClient', () => {
     });
   });
 
+  describe('requestInterceptor', () => {
+    it('can intercept out going requests', async () => {
+      let expectedBody: Record<string, unknown>;
+
+      const client = new WebClient(token, {
+        requestInterceptor: (config: RequestConfig) => {
+          expectedBody = Object.freeze({
+            method: config.method,
+            base_url: config.baseURL,
+            path: config.url,
+            body: config.data ?? {},
+            query: config.params ?? {},
+            headers: structuredClone(config.headers),
+            test: 'static-body-value',
+          });
+          config.data = expectedBody;
+
+          config.headers.test = 'static-header-value';
+          config.headers['Content-Type'] = 'application/json';
+
+          return config;
+        },
+      });
+
+      nock('https://slack.com/api', {
+        reqheaders: {
+          test: 'static-header-value',
+          'Content-Type': 'application/json',
+        },
+      })
+        .post(/method/, (requestBody) => {
+          expect(requestBody).to.deep.equal(expectedBody);
+          return true;
+        })
+        .reply(200, (_uri, requestBody) => {
+          expect(requestBody).to.deep.equal(expectedBody);
+          return { ok: true, response_metadata: requestBody };
+        });
+
+      await client.apiCall('method');
+    });
+  });
+
+  describe('adapter', () => {
+    it('allows for custom handling of requests', async () => {
```

Comment:
This test is insufficient for me to understand the difference between request interceptor and adapter. Let's discuss this feature some more. Why would web-api users want to use adapters over a request interceptor?

Reply:
In some cases a web-api user will already have a preconfigured client which they would like to use. They may have a predefined axios instance configured with their desired settings to be used with their internal proxy for all external requests. This client is shared across their project, and is re-usable via their dependency injection framework. Their custom http client may have the following properties:
    - instance already setup to use interceptor for tracing (datadog, new relic, etc.)
    - instance already configured with desired timeout, retries etc.
    - instance may be using using a SDK configured with their internal proxy OpenAPI specs


Yes there is overlap in functionality, but interceptor and adaptor can apply for different use cases. For client who do not have a custom axios instance they want to use, they can just setup an interceptor.

Some user's may want to configure both an interceptor and an adapter based on their way of thinking. For example:
- configure a custom interceptor to shape the desired request
- configure a custom adapter to ensure that client with all desired configuration is in use

Reply:
Note that `octokit` and `datadog` supports this in their on way.  Here is some pseudo code:

For example with [octokit](https://octokit.github.io/rest.js/v21/#hooks) can do the following

```ts
octokit.hook.wrap('request', async (request, options) => {
  const proxyRequest = transformOutgoinRequests(options)
  const proxyResponse = await customAxiosClient.request(proxyRequest, {
    konnectAccessToken: authToken,
  })

  return proxyResponse
})
```

For [datadog](https://github.com/DataDog/datadog-api-client-typescript/blob/c6d3465bd0aeb8545facee8892a4a30e534cfc46/packages/datadog-api-client-common/http/isomorphic-fetch.ts#L13)
 
 
```ts
export class CustomProxyHttpLib implements DatadogHttpLibrary {
  constructor(
    private readonly customAxiosClient,
  ) {}

  public send(request: client.RequestContext): Promise<client.ResponseContext> {
    const proxyRequest = transformOutgoinRequests(request)
    const proxyResponse = await this.customAxiosClient.request(proxyRequest, {
      konnectAccessToken: authToken,
    })
    
    return transformToDDResponseInterface(proxyResponse)
  }
}
```

Reply:
Thanks for taking the time to explain that! That sounds fine and reasonable to me, esp. if these are already well-established and leveraged axios features.

However, this definitely solidifies the need, in my opinion, to add hand-written docs for this feature as part of the deliverable. Other users will love that, but perhaps you as the PR author will not (as it's more work) 😅 I think adding a section covering request interceptors and adapters to the document under `docs/content/packages/web-api.md`. I'd suggest a new section after the "Custom API URL" section (around line 800 in that document). Doesn't have to be long (you can see the other sections in this document are typically one code example and a couple of paragraphs). In fact, I think what you wrote in your comment above is a great start for such docs content. Couple that with links to the relevant axios docs should be plenty to have this featured covered from a documentation perspective.

Reply:
I've adjusted the test to better reflect real world use case and added documentation under `docs/content/packages/web-api.md`